composer.json

```diff
@@ -23,6 +23,7 @@
         "ext-openssl": "*",
         "ext-session": "*",
         "ext-tokenizer": "*",
+        "ext-zlib": "*",
```

Comment:
Adding an additional extension in a patch release is a breaking change. This should either only be in the `suggest` section (with an `extension_loaded` check in the code) or be targeted at 12.x instead.

Reply:
Just for reference, this isn't a breaking change. Composer will warn you about the missing extension. Adding new dependencies on patch releases is allowed.

Reply:
@driesvints thanks for clarifying that. I understand the argument for composer dependencies, but does it also applies for extensions?

That Composer warns about a missing extension, it does, indeed, all good about that.

What I mean is: if someone is running a Laravel 11 app on a server which they are not allowed to freely install extensions, such as shared hosting, then they wouldn't be able to update their app due to the inability to install a newly required extension.

Doesn't it count as a breaking change? As a requirement change?

I believe a `suggests` entry would be a better fit in that case.

If that case is still allowed, as a requirement, I believe docs should also be updated to reflect the newly required extension on the "Server Requirements" section:

https://laravel.com/docs/11.x/deployment#server-requirements

Reply:
If the extension isn't needed for a core essential of Laravel then it can indeed be added as a suggest. I believe this can be done here since the feature is indeed isolated.

SemVer doesn't look at things like shared hosting and the ability to not install extensions. It's solely focused on what is a breaking change and adding a dependency on a patch release isn't one. A good dependency manager, like Composer, will warn you about the new requirement and prevent you from moving forward in a deploy/install process until you've fulfilled the requirement. And thus nothing will break.

Reply:
@rmunate move this to suggest please. Also for the `composer.json` of the support package.